src/metabase/driver/common.clj

```diff
@@ -435,16 +435,18 @@
 (def ^:private ^clojure.lang.PersistentVector days-of-week
   [:monday :tuesday :wednesday :thursday :friday :saturday :sunday])
 
-(defn start-of-week->int
-  "Returns the int value for the current :start-of-week setting value, which ranges from 0 (:monday) to 6 (:sunday).
-  If the :start-of-week setting does not have a value, then `nil` is returned."
+(s/defn start-of-week->int :- (s/pred (fn [n] (and (integer? n) (<= 0 n 6)))
+                                      "Start of week integer")
+  "Returns the int value for the current [[metabase.public-settings/start-of-week]] Setting value, which ranges from
+  `0` (`:monday`) to `6` (`:sunday`). This is guaranteed to return a value."
```

Comment:
Docstring was off a little here. This never returns `nil`. It returns the default value `6` for `:sunday` if `start-of-week` is not set.

Reply:
If that's the case should we get rid of the `when-let`? Structurally, this still seems like it is ready to return nil.

Reply:
Yeah that's true. I'll fix it

Reply:
fixed